### PR TITLE
Service modifications support for ES6 classes

### PR DESCRIPTION
**Pull request by akazakou**
## :memo: Description

In some cases architectual solutions can split service and service runner logic. In that case, if you need provide to service some settings or update schema parameters, you can use `schemaMods`. 

It working for GenericObject definition, but not working for ES6 classes. I added providing `schemaMods` parameters to ES6 classes.

### :dart: Relevant issues
#483 Service modifications support for ES6 classes

### :gem: Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### :scroll: Example code
```js
class ES6Service extends Service {
	constructor(broker, schemaMods) {
		super(broker);

		this.name = "es6-with-schema-mods";
		this.version = 2;
		this.settings = {};
		this.actions = {
			send: () => Prmise.resolve()
		};

		if (schemaMods && schemaMods.settings && schemaMods.settings.connectionString) {
			this.settings.connectionString = schemaMods.settings.connectionString;
		}

		// this should prevent override this.actions in parseServiceSchema method
		this.parseServiceSchema(Object.assign({}, this));
	}
}

let broker = new ServiceBroker({ logger: false });
broker.createService(ES6Service, { settings: { connectionString: "mongodb://localhost:27017/test" } });
``` 

## :vertical_traffic_light: How Has This Been Tested?

I've added 2 additional unit tests to check schema creating proccess for ES6 classes

- [x] should create service from ES6 instance without schema mods
- [x] should create service from ES6 instance with schema mods

## :checkered_flag: Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] **I have added tests that prove my fix is effective or that my feature works**
- [x] **New and existing unit tests pass locally with my changes**
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/service.js [hunk 1]
```diff
@@ -39,11 +39,11 @@ class Service {
 	 * Creates an instance of Service by schema.
 	 *
 	 * @param {ServiceBroker} 	broker	broker of service
-	 * @param {Object} 			schema	schema of service
+	 * @param {Object=} 		schema	schema of service
 	 *
 	 * @memberof Service
 	 */
-	constructor(broker, schema) {
+	constructor(broker, schema = undefined) {
```
Comment (icebob) [inline src/service.js]: Why need it `schema = undefined`. I think it is not relevant, because if you don't define˙it, it will be `undefined`

Comment (akazakou) [reply src/service.js]: My IDE highlit this. I just added this for comfortable IDE workaround. If it not required, I can remove this

Comment (icebob) [reply src/service.js]: OK.

Comment (akazakou) [reply src/service.js]: @icebob I reverted changes in `service.js` file. Right now only related to feature changes

**Pull request merged**